src/main/java/org/primefaces/component/selectonemenu/SelectOneMenuRenderer.java

```diff
@@ -429,19 +429,21 @@ protected void encodeOptionsAsTable(FacesContext context, SelectOneMenu menu, Li
             else {
                 for (int j = 0; j < columns.size(); j++) {
                     Column column = columns.get(j);
```

Comment:
A better fix would be...
```
Column column = columns.get(j);
if (!column.isRendered) {
   continue;
}
```

Reply:
That's a nice suggestion. I refactored my PR.